tidb/src/main/java/io/tidb/bigdata/tidb/allocator/DynamicRowIDAllocator.java

```diff
@@ -93,10 +160,26 @@ private void checkUpdate() {
     }
   }
 
-  public long getSharedRowId() {
+  private void updateCurrentShard() {
+    if (type != RowIDAllocatorType.AUTO_RANDOM) {
+      return;
+    }
+
+    Random random = RANDOM_THREAD_LOCAL.get();
+    // Use MurmurHash3 to make sure the bits of shardSeed >= 15
+    currentShardSeed = MurmurHash3.hash32(random.nextLong());
+  }
+
+  public long getImplicitRowId() {
     checkUpdate();
     index++;
-    return RowIDAllocator.getShardRowId(maxShardRowIDBits, index, index + start);
+    return RowIDAllocator.getShardRowId(shardBits, index, index + start, isUnsigned);
```

Comment:
maybe`getShardRowId(shardBits, index, index + start, false)` will be  clearer

Reply:
There is a potential of bug if we change the logic for ImplicitRowId and lots of code needs to know if it is unsigned
So it's better to set isUnsigned in method isUnsigned() together rather than hardcode separately

Reply:
it makes sense